### ISSUE TITLE
How to generate segmentation labels for tusimple data？

### ISSUE THREAD

**Issue by daigang896** [voldemortX/pytorch-auto-drive#40]
Hello，
    How to generate segmentation labels for tusimple data？Because of the self generated label training, the accuracy ACC is very low, only 0.665.  Use：python main_landec.py --epochs=50 --lr=0.2 --batch-size=20 --dataset=tusimple --method=scnn --backbone=erfnet --mixed-precision --exp-name=erfnet_scnn_tusimple. Only segmentation labels are different. Is there a problem? Hope to provide some help?

**Comment by voldemortX** [comment]
@daigang896  Hi, do you mean 66.5%? We use the SAD version labels with 6 lane lines, the generation scripts should be originated from [SCNN](https://github.com/XingangPan/seg_label_generate). Did you download labels according to [this](https://github.com/voldemortX/pytorch-auto-drive/blob/master/docs/datasets/TUSIMPLE.md)?

**Comment by daigang896** [comment]
Hello, I can't download the segmentation labels. In addition, I want to be able to generate this segmentation labels myself.  Let me first look at the generated script to see if it is different from what I wrote.

**Comment by voldemortX** [comment]
@daigang896 Do you need the downloads in BaiduDrive?

**Comment by daigang896** [comment]
Yes, thank you. In addition, I need python code to generate this segmentation labels. I wonder if you can provide it.

**Comment by voldemortX** [comment]
> Yes, thank you. In addition, I need python code to generate this segmentation labels. I wonder if you can provide it.

BaiduDrive link:
https://pan.baidu.com/s/1VWI0a1oSp16g1lSFQQO4Jg 
code: vydo

However, I don't think there is reliable python code for generating seg labels, you may have to work with cpp for now.

**Comment by daigang896** [comment]
Well, thank you.

**Comment by daigang896** [comment]
Hello, I read this discussion. The Python code(https://github.com/cardwing/Codes-for-Lane-Detection/files/2642182/segLabelGen0.zip) in this discussion is incomplete. I'll find other codes.

**Comment by voldemortX** [comment]
Well.  that's about the only python code I can find.  Do post here if you find better python codes. Cheers.

**Comment by daigang896** [comment]
Well.I try to find more complete code. If I can find it, I will send the link here.

**Comment by daigang896** [comment]
I found a code link to generate the segmentation labels for tusimple data. Thanks for the code released by ZJULearning. The link is: https://github.com/ZJULearning/resa/blob/main/tools/generate_seg_tusimple.py.

**Comment by voldemortX** [comment]
@daigang896 That looks good! By a rough look it seems the same algorithm as the cpp version.

**Comment by daigang896** [comment]
OK, this is a complete python code and can be run easily.

**Comment by voldemortX** [comment]
@cedricgsh I'll add links for generation scripts in the doc. Thanks for your investigations!

**Comment by voldemortX** [comment]
@daigang896 I've summarized what we found in this issue and updated the doc b21cb07

I'll close this issue for now since the problem seems resolved, feel free to open a Pull Request or Issue for feature request to add generation scripts for this repo. Thanks again for your contributions.
